refactor: create FlavorName component

uncenter opened the pull request.

andreasgrafen commented:
Oh! Putting the flavors into a separate component is a good call.

I'm not sure about having them `strong` per default as they might not be something we semantically want to draw attention in every usecase – for example a 4-way color switcher later on.

For the palette page I feel like it's also sensible to keep the different tables hotlinkable with the ID argument as this allows us to easily direct users to the flavor directly.

uncenter commented:
> Oh! Putting the flavors into a separate component is a good call.
> 
> I'm not sure about having them `strong` per default as they might not be something we semantically want to draw attention in every usecase – for example a 4-way color switcher later on.
> 
> For the palette page I feel like it's also sensible to keep the different tables hotlinkable with the ID argument as this allows us to easily direct users to the flavor directly.

I've incorporated your suggestions and updated the branch if you are still interested.
